pkg/build/registry/build/strategy.go

```diff
@@ -140,3 +147,20 @@ func (detailsStrategy) AllowUnconditionalUpdate() bool {
 
 // DetailsStrategy is the strategy used to manage updates to a Build revision
 var DetailsStrategy = detailsStrategy{Strategy}
+
+type statusStrategy struct {
+	strategy
+}
+
+// StatusStrategy is the strategy used to manage build status updates
+var StatusStrategy = statusStrategy{Strategy}
+
+func (statusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newBuild := obj.(*api.Build)
+	oldBuild := old.(*api.Build)
+	newBuild.Spec = oldBuild.Spec
```

Comment:
Is allowing the build annotations to be mutated post creation dangerous from a security perspective?